hedera-node/src/main/java/com/hedera/services/state/logic/StandardProcessLogic.java

```diff
@@ -86,42 +86,64 @@ public StandardProcessLogic(
 	public void incorporateConsensusTxn(SwirldTransaction platformTxn, Instant consensusTime, long submittingMember) {
 		try {
 			final var accessor = expandHandleSpan.accessorFor(platformTxn);
-			Instant effectiveConsensusTime = consensusTime;
-			if (accessor.canTriggerTxn()) {
-				final var offset = dynamicProperties.triggerTxnWindBackNanos();
-				effectiveConsensusTime = consensusTime.minusNanos(offset);
-			}
 
-			if (!invariantChecks.holdFor(accessor, effectiveConsensusTime, submittingMember)) {
+			if (!invariantChecks.holdFor(accessor, consensusTime, submittingMember)) {
 				return;
 			}
 
-			sigImpactHistorian.setChangeTime(effectiveConsensusTime);
-			expiries.purge(effectiveConsensusTime.getEpochSecond());
+			consensusTimeTracker.reset(consensusTime);
+
+			sigImpactHistorian.setChangeTime(consensusTime);
+			expiries.purge(consensusTime.getEpochSecond());
 			sigImpactHistorian.purge();
 			recordStreaming.resetBlockNo();
 
-			doProcess(submittingMember, consensusTime, effectiveConsensusTime, accessor);
+			doProcess(submittingMember, consensusTimeTracker.isFirstUsed()
+					? consensusTimeTracker.nextTransactionTime(true)
+					: consensusTimeTracker.firstTransactionTime(), accessor);
+
+			processScheduledTransactions(consensusTime, submittingMember);
```

Comment:
It seems almost inconceivable the full call stack `processScheduledTransactions -> triggerNextTransactionExpiringAsNeeded -> expire -> advanceCurrentMinSecond` will be inlined, so I would definitely wrap this in a,
```
if (networkCtxManager.currentTxnIsFirstInConsensusSecond()) {
    processScheduledTransactions(consensusTime, submittingMember);
}
```

Reply:
I just don't want to have to remember to come back to this if it is possbile to schedule more than a couple of hundred transactions per second ;-). 

Would it be ok to just directly check schedules.getCurrentMinSecond() via a single call to a new function in schedulestore?

Reply:
I added shouldProcessScheduledTransactions to ScheduleProcessing, which should be much faster. It probably won't get in-lined, but I think this is a good middle ground?

Reply:
Definitely lower-hanging fruit to pick given that change! Sounds good to me.